build-logic/publishing/src/main/kotlin/buildlogic/cleanupMavenPom.kt

```diff
@@ -0,0 +1,24 @@
+package buildlogic
+
+import org.gradle.api.publish.maven.MavenPom
+
+fun MavenPom.cleanupMavenPom() {
```

Comment:
I'm not sure I understand the point of this cleanup method?

Reply:
By default, Gradle produces pom files with `<scope>compile</scope>` which is default in Maven, and it somewhat makes the pom noisy.
The second change is that it removes `dependencyManagement` section completely.
In practice, that dependency management section does **nothing**, except it might confuse the users.

For instance, here's the pom for `sigstore-java:0.1.0`: https://repo1.maven.org/maven2/dev/sigstore/sigstore-java/0.1.0/sigstore-java-0.1.0.pom
It does have `<dependencyManagement>`, however, the section will never be used by Maven, and it will never be used by Gradle.
So I am inclined it is worth just removing the section to avoid confusing the users.

Reply:
I'm not super familiar with bom mechanics, but `<dependencyManagement>` is not transitive? I guess only if you apply this with `platform`?

Reply:
With Maven, the only way to use dependencyManagement is to use scope=import, however even then pom.xml that declares management should have type=bom: https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope.

In other words, for jar artifacts dependencyManagement is never used

Reply:
I've removed `cleanupMavenPom` for now.